src/payment/strategies/adyenv2/adyenv2-payment-strategy.spec.ts

```diff
@@ -0,0 +1,784 @@
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledWith({
+                methodId: 'authorizenet',
+                paymentData: {
+                    nonce: JSON.stringify(getValidChallengeResponse(), null, 2),
+                },
+            });
+        });
+
+        it('throws an error when payment is not set properly into payload', () => {
+            const payload = {
+                payment: undefined,
+            };
+
+            expect(() => strategy.execute(payload))
+                .toThrow(PaymentArgumentInvalidError);
+        });
+
+        it('does not post 3ds data to Adyen if 3ds is not enabled', async () => {
+            const response = new RequestError(getResponse(getErrorPaymentResponseBody()));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValue(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, response)));
+
+            try {
+                await strategy.execute(getOrderRequestBody());
+            } catch (error) {
+                expect(error).toBeInstanceOf(RequestError);
+                expect(formPoster.postForm).not.toHaveBeenCalled();
+            }
+        });
+
+        it('throws error if order is missing', async () => {
+            const state = store.getState();
+
+            jest.spyOn(state.order, 'getOrder')
+                .mockReturnValue(null);
+
+            try {
+                await strategy.finalize();
+            } catch (error) {
+                expect(error).toBeInstanceOf(OrderFinalizationNotRequiredError);
+            }
+        });
+
+        it('returns 3DS2 IndentifyShopper flow', async () => {
+            const error = new RequestError(getResponse({
+                ...getErrorPaymentResponseBody(),
+                ...getIdentifyShopperErrorResponse(),
+            }));
+
+            jest.spyOn(adyenCheckout, 'create')
+                .mockImplementationOnce(jest.fn(() => adyenComponent))
+                .mockImplementationOnce(jest.fn((_type, options) => {
+                    componentOptions = options;
+
+                    return threeDS2Component;
+                }));
+
+            jest.spyOn(adyenV2ScriptLoader, 'load').mockReturnValue(Promise.resolve(adyenCheckout));
+
+            jest.spyOn(paymentActionCreator, 'submitPayment')
+                .mockReturnValueOnce(of(createErrorAction(PaymentActionType.SubmitPaymentFailed, error)))
+                .mockReturnValueOnce(submitPaymentAction);
+
+            await strategy.initialize(options);
+            await strategy.execute(getOrderRequestBody());
+
+            expect(paymentActionCreator.submitPayment).toHaveBeenCalledTimes(2);
+            expect(paymentActionCreator.submitPayment).toHaveBeenLastCalledWith({
+                methodId: 'authorizenet',
+                paymentData: {
+                    nonce: JSON.stringify({
+                        threeDS2Token: 'token' ,
```

Comment:
Remove this space before the comma.
```suggestion
                        threeDS2Token: 'token',
```